llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -1650,10 +1650,11 @@ size_t SYCLLowerESIMDPass::runOnFunction(Function &F,
   // There is a current limitation of GPU vector backend that requires kernel
   // functions to be inlined into the kernel itself. To overcome this
   // limitation, mark every function called from ESIMD kernel with
-  // 'alwaysinline' attribute.
+  // 'alwaysinline' attribute, except few cases.
```

Comment:
Can  you please add one statement explaining why VCFunctionMD::VCStackCall should not be inlined?
Description and comment say nothing about reasoning, which may be permanent, or may become obsolete some day, it is unclear.

Reply:
ok, added comments for all the cases where we don't add `alwaysinline`

Reply:
Great! Thank you!